stl/inc/functional

```diff
@@ -1557,24 +1557,17 @@ constexpr auto _Call_front_binder(index_sequence<_Ix...>, _Cv_FD&& _Obj, _Cv_tup
 template <class _Fx, class... _Types>
 class _Front_binder { // wrap bound callable object and arguments
 private:
-    static_assert(is_constructible_v<decay_t<_Fx>, _Fx>,
-        "std::bind_front() requires the decayed callable to be constructible from an undecayed callable");
-    static_assert(is_move_constructible_v<decay_t<_Fx>>,
-        "std::bind_front() requires the decayed callable to be move constructible");
-    static_assert(conjunction_v<is_constructible<decay_t<_Types>, _Types>...>,
-        "std::bind_front() requires the decayed bound arguments to be constructible from undecayed bound arguments");
-    static_assert(conjunction_v<is_move_constructible<decay_t<_Types>>...>,
-        "std::bind_front() requires the decayed bound arguments to be move constructible");
+    using _Seq = index_sequence_for<_Types...>;
 
-    using _Seq    = index_sequence_for<_Types...>;
-    using _First  = decay_t<_Fx>;
-    using _Second = tuple<decay_t<_Types>...>;
+    _Compressed_pair<_Fx, tuple<_Types...>> _Mypair;

```

Comment:
This data member change appears to be ABI-safe because you're preventing non-decayed types from compiling here, and for decayed types there is no difference. Definitely one of the more risky transformations, but seems fine. No change requested.

Reply:
Yes, it's subtle here but safe.